test/core/tsi/ssl_transport_security_utils_test.cc

```diff
@@ -429,6 +445,184 @@ INSTANTIATE_TEST_SUITE_P(FrameProtectorUtil, FlowTest,
+  absl::StatusOr<Slice> issuer_slice = LoadFile(kIntermediateCrlIssuer, false);
+  ASSERT_EQ(issuer_slice.status(), absl::OkStatus());
+  X509_CRL* crl = read_crl(crl_slice->as_string_view());
+  X509* issuer = read_cert(issuer_slice->as_string_view());
+  EXPECT_EQ(VerifyCrlSignature(crl, issuer), 1);
+}
+
+TEST(CrlUtils, VerifySignatureModifiedSignature) {
+  absl::StatusOr<Slice> crl_slice = LoadFile(kModifiedSignature, false);
+  ASSERT_EQ(crl_slice.status(), absl::OkStatus()) << crl_slice.status();
+  absl::StatusOr<Slice> issuer_slice = LoadFile(kCrlIssuer, false);
+  ASSERT_EQ(issuer_slice.status(), absl::OkStatus());
+  X509_CRL* crl = read_crl(crl_slice->as_string_view());
+  X509* issuer = read_cert(issuer_slice->as_string_view());
+  EXPECT_EQ(VerifyCrlSignature(crl, issuer), 0);
+}
+
+TEST(CrlUtils, VerifySignatureModifiedContent) {
+  absl::StatusOr<Slice> crl_slice = LoadFile(kModifiedContent, false);
+  ASSERT_EQ(crl_slice.status(), absl::OkStatus()) << crl_slice.status();
+  absl::StatusOr<Slice> issuer_slice = LoadFile(kCrlIssuer, false);
+  ASSERT_EQ(issuer_slice.status(), absl::OkStatus());
+  X509_CRL* crl = read_crl(crl_slice->as_string_view());
+  EXPECT_EQ(crl, nullptr);
+}
+
+TEST(CrlUtils, VerifySignatureWrongIssuer) {
+  absl::StatusOr<Slice> crl_slice = LoadFile(kValidCrl, false);
+  ASSERT_EQ(crl_slice.status(), absl::OkStatus()) << crl_slice.status();
+  absl::StatusOr<Slice> issuer_slice = LoadFile(kIntermediateCrlIssuer, false);
+  ASSERT_EQ(issuer_slice.status(), absl::OkStatus());
+  X509_CRL* crl = read_crl(crl_slice->as_string_view());
+  X509* issuer = read_cert(issuer_slice->as_string_view());
+  EXPECT_EQ(VerifyCrlSignature(crl, issuer), 0);
+}
+
+TEST(CrlUtils, VerifySignatureWrongIssuer2) {
+  absl::StatusOr<Slice> crl_slice = LoadFile(kIntermediateCrl, false);
+  ASSERT_EQ(crl_slice.status(), absl::OkStatus()) << crl_slice.status();
+  absl::StatusOr<Slice> issuer_slice = LoadFile(kCrlIssuer, false);
+  ASSERT_EQ(issuer_slice.status(), absl::OkStatus());
+  X509_CRL* crl = read_crl(crl_slice->as_string_view());
+  X509* issuer = read_cert(issuer_slice->as_string_view());
+  EXPECT_EQ(VerifyCrlSignature(crl, issuer), 0);
+}
+
+TEST(CrlUtils, VerifySignatureNullCrl) {
+  absl::StatusOr<Slice> issuer_slice = LoadFile(kCrlIssuer, false);
+  ASSERT_EQ(issuer_slice.status(), absl::OkStatus());
+  X509_CRL* crl = nullptr;
+  X509* issuer = read_cert(issuer_slice->as_string_view());
+  EXPECT_EQ(VerifyCrlSignature(crl, issuer), -1);
+}
+
+TEST(CrlUtils, VerifySignatureNullCert) {
+  absl::StatusOr<Slice> crl_slice = LoadFile(kIntermediateCrl, false);
+  ASSERT_EQ(crl_slice.status(), absl::OkStatus()) << crl_slice.status();
+  X509_CRL* crl = read_crl(crl_slice->as_string_view());
+  X509* issuer = nullptr;
+  EXPECT_EQ(VerifyCrlSignature(crl, issuer), -1);
+}
+
+TEST(CrlUtils, VerifySignatureNullCrlAndCert) {
+  X509_CRL* crl = nullptr;
+  X509* issuer = nullptr;
+  EXPECT_EQ(VerifyCrlSignature(crl, issuer), -1);
```

Comment:
nit: Can we simplify this to 1 line?
```
EXPECT_EQ(VerifyCrlSignature(/*crl=*/nullptr, /*issuer=*/nullptr), -1);
```

Reply:
Same as my previous comment about that potentially making it _less_ clear what is being asserted/tested